AUTHORING_LIBRARIES.md

```diff
@@ -0,0 +1,58 @@
+# Authoring Libraries
+
+Authoring libraries with rollup, using this blueprint, requires some more explicit conventions than what folks pre-V2-addons may be used to.
+
+## Building
+
+This blueprint uses [Rollup](https://rollupjs.org/) for assembling and transpiling your code in to a native npm package that can be imported from anywhere[^anywhere].
+
+There are a few reasons we need a build step:
+- use of pre-shipped JS features (decorators, and other in-progress ECMA proposals)
+- co-located components are not modules-by-default[^gjs]
+- building at publish time makes apps faster -- because the apps then have less to transform themselves[^embroider-compat-v2]
+
+If you'd like to provide build-time behaviors, using [unplugin](https://github.com/unjs/unplugin) allows you to write a build-time plugin once, using similar-to-rollup-APIs, and works for Vite, Rollup, Webpack, and ESBuild.
```

Comment:
as this is great to have moving forward, is this something we can teach already?

Reply:
It'll make more sense once embroider has inverted the build tool. 
Right now you can use unplugin in embroider, but due to current-embroider being a broccoli-plugin around webpack, we lose the built in "watch mode" that unplugins can augment (addWatchFile?, I think is the API)

Reply:
got it! Thanks for clarification!